Firestore/core/src/firebase/firestore/core/sync_engine.cc

```diff
@@ -110,10 +110,23 @@ ViewSnapshot SyncEngine::InitializeViewAndComputeSnapshot(const Query& query,
   DocumentMap docs = local_store_->ExecuteQuery(query);
   DocumentKeySet remote_keys = local_store_->GetRemoteDocumentKeys(target_id);
 
+  // If there are already queries mapped to the target id, create a synthesized
+  // target change to apply the sync state from those queries to the new query.
+  auto current_sync_state = SyncState::None;
+  TargetChange synthesized_current_change;
```

Comment:
This isn't equivalent to the android version. There this defaults to null, and only if there's a synthesized change does it get set to a non-null value.

Here you're creating a default TargetChange. This type should likely be `absl::optional<TargetChange>` (which is the type of the parameter to `ViewChange::ApplyChanges`.

Reply:
Done.